cmd/icingadb/main.go

```diff
@@ -67,9 +68,17 @@ func run() int {
 
 	heartbeat := icingaredis.NewHeartbeat(ctx, rc, logger)
 	ha := icingadb.NewHA(ctx, db, heartbeat, logger)
+
 	// Closing ha on exit ensures that this instance retracts its heartbeat
 	// from the database so that another instance can take over immediately.
-	defer ha.Close()
+	{
+		defer func() {
+			ctx, cancelCtx := context.WithTimeout(context.Background(), 3*time.Second)
+			ha.Close(ctx)
+			cancelCtx()
+		}()
+	}
```

Comment:
What's the point of these curly braces? That defer statement won't pollute the surrounding scope with any names or something like that.

Reply:
That was an 🐘.